components/script/dom/htmlcanvaselement.rs

```diff
@@ -407,15 +407,23 @@ impl HTMLCanvasElement {
         &self,
         image_type: &EncodedImageType,
         quality: Option<f64>,
-        bytes: &[u8],
+        snapshot: &Snapshot,
         encoder: &mut W,
     ) {
+        // We can't use self.Width() or self.Height() here, since the size of the canvas
+        // may have changed since the canvas was created. Truncating the dimensions to a
+        // u32 can't panic, since the data comes from a canvas which is always smaller than
+        // u32::MAX.
+        let canvas_data = snapshot.data();
+        let width = snapshot.size().width as u32;
+        let height = snapshot.size().height as u32;
```

Comment:
Reading this reminds me that we are pretty chaotic about canvas size type (u32 vs u64). We should unify this (by using u32 evrywhere) and also make it reflect the snapshot.